File: articles/simulation-and-computational-modeling/fortran/resource_simulation.f90
program resource_simulation
  implicit none

  integer, parameter :: steps = 20
  real(8) :: stock
  real(8), parameter :: growth_rate = 0.18d0
  real(8), parameter :: capacity = 100.0d0
  real(8), parameter :: extraction = 6.0d0
  real(8) :: growth
  integer :: t

  stock = 70.0d0

  print '(A)', 'step resource_stock'

  do t = 0, steps
    print '(I0,1X,F10.4)', t, stock
    growth = growth_rate * stock * (1.0d0 - stock / capacity)
    stock = max(0.0d0, stock + growth - extraction)
  end do
end program resource_simulation

File: articles/simulation-and-computational-modeling/cpp/resource_simulation.cpp
#include <algorithm>
#include <iomanip>
#include <iostream>

int main() {
    double stock = 70.0;
    const double growth_rate = 0.18;
    const double capacity = 100.0;
    const double extraction = 6.0;
    const int steps = 20;

    std::cout << "step,resource_stock\n";

    for (int step = 0; step <= steps; ++step) {
        std::cout << step << "," << std::fixed << std::setprecision(6) << stock << "\n";
        const double growth = growth_rate * stock * (1.0 - stock / capacity);
        stock = std::max(0.0, stock + growth - extraction);
    }

    return 0;
}
